src/lib/Repository/UiFieldGroupService.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Repository;
+
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\Core\Helper\FieldsGroups\FieldsGroupsList;
+use eZ\Publish\API\Repository\Values\ContentType\FieldDefinition;
+use EzSystems\HybridPlatformUi\Repository\Values\ContentType\UiFieldGroup;
+
+/**
+ * Service for loading Field groups.
+ */
+class UiFieldGroupService
+{
+    /**
+     * @var ContentTypeService
+     */
+    private $contentTypeService;
+    /**
+     * @var FieldsGroupsList
+     */
+    private $fieldsGroupsList;
+
+    public function __construct(ContentTypeService $contentTypeService, FieldsGroupsList $fieldsGroupsList)
+    {
+        $this->contentTypeService = $contentTypeService;
+        $this->fieldsGroupsList = $fieldsGroupsList;
+    }
+
+    /**
+     * Loads field groups for a piece of content and includes the group name.
+     *
+     * @param ContentInfo $contentInfo
+     *
+     * @return UiFieldGroup[]
+     */
+    public function loadFieldGroups(ContentInfo $contentInfo)
+    {
+        $contentType = $this->contentTypeService->loadContentType($contentInfo->contentTypeId);
+        $fields = $contentType->getFieldDefinitions();
+
+        $fieldGroups = [];
+        foreach ($this->fieldsGroupsList->getGroups() as $groupId => $groupName) {
+            $fieldsInGroup = array_filter($fields, function (FieldDefinition $field) use ($groupId) {
```

Comment:
Would you mind refactoring this part to a private method ? It has quite a lot of logic and not much explanations. Something like `$fieldsInGroup = filterFieldsFromGroup($fields, $groupId);`.